packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -255,18 +257,42 @@ fn process_fn_items(
         None => quote! {},
     };
 
-    let contract = match &manifest.contract_id {
-        Some(contract_id) => {
+    let contracts = match &manifest.contract_id {
+        ContractIds::Single(contract_id) => match contract_id {
+            Some(contract_id) => {
+                quote! {
+                    let manifest_contract_id = Bech32ContractId::from_str(#contract_id).expect("Failed to parse manifest 'contract_id' as Bech32ContractId");
+                    let receipt_contract_id = Bech32ContractId::from(id);
+                    if receipt_contract_id != manifest_contract_id {
+                        Logger::info("Not subscribed to this contract. Will skip this receipt event. <('-'<)");
+                        continue;
+                    }
+                }
+            }
+            None => quote! {},
+        },
+        ContractIds::Multiple(contract_ids) => {
+            let contract_ids_str: Vec<String> = contract_ids
+                .iter()
+                .filter_map(|id| id.as_ref().cloned())
+                .collect();
+
             quote! {
-                let manifest_contract_id = Bech32ContractId::from_str(#contract_id).expect("Failed to parse manifest 'contract_id' as Bech32ContractId");
                 let receipt_contract_id = Bech32ContractId::from(id);
-                if receipt_contract_id != manifest_contract_id {
+                let contract_ids_set: HashSet<Bech32ContractId> = {
```

Comment:
There should be a way; you could do a map across the list of contract ID strings, turn it back into an iter, and pass it into `HashSet::from_iter()`. I played around with this to see the set could be constructed at compile time instead of at run time, but it looks like `ToTokens` isn't implemented for `HashSet<Bech32ContractId>`. 🤷🏽‍♂️

Reply:
Yeah I also felt this was a little janky, I tried to write it initially in a more readable fashion, but the compiler led me into this direction for reasons I can't exactly recall. Let me have another crack at this block.

Reply:
Yeah I've been wrestling with this for a bit now. The only way is to implement `ToTokens` with a custom type wrapper like: 
```rust
pub struct Bech32ContractIdSet(HashSet<Bech32ContractId>);
```
The ToTokens custom method seems like a lot of code for a small bit of functionality, whereas what we have here currently is compiling with no issues. If we think, implementing this method will be useful down the line, I am happy to implement this.

Reply:
@rvmelkonian So I don't believe it requires you to implement `ToTokens`.

```rust
        ContractIds::Multiple(contract_ids) => {
            let contract_ids = contract_ids
                .iter()
                .map(|id| {
                    quote! {
                        Bech32ContractId::from_str(#id)
                            .expect("Failed to parse manifest 'contract_id' as Bech32ContractId")
                    }
                })
                .collect::<Vec<proc_macro2::TokenStream>>();

            quote! {
                let receipt_contract_id = Bech32ContractId::from(id);
                let contract_ids = HashSet::from([#(#contract_ids),*]);

                if !contract_ids.contains(&receipt_contract_id) {
                    Logger::info("Not subscribed to this contract. Will skip this receipt event. <('-'<)");
                    continue;
                }
            }
        }
```

Doesn't have to be this ☝🏽 but something like this is more so what I was referring to. 

Which produces:

```rust
                    Receipt::Call {
                        id: contract_id, amount, asset_id, gas, param1, to: id, .. }
                        => {
                        let receipt_contract_id = Bech32ContractId::from(id);
                        let contract_ids =
                            HashSet::from([Bech32ContractId::from_str("fuel1t8v0g6h2jcnjtlpa0p3z244qcqjc8tp2g6fuf63uc8jam6e4j4uqen2kpu").expect("Failed to parse manifest 'contract_id' as Bech32ContractId"),
                                        Bech32ContractId::from_str("fuel18hchrf7f4hnpkl84sqf8k0sk8gcauzeemzwgweea8dgr7eachv4s86r9t9").expect("Failed to parse manifest 'contract_id' as Bech32ContractId")]);
                        if !contract_ids.contains(&receipt_contract_id) {
                                Logger::info("Not subscribed to this contract. Will skip this receipt event. <('-'<)");
                                continue;
                            }
                        let fn_name = decoder.selector_to_fn_name(param1);
                        return_types.push(param1);
                        callees.insert(id);
                        let data =
                            bincode::serialize(&abi::Call {
                                            contract_id,
                                            to: id,
                                            amount,
                                            asset_id,
                                            gas,
                                            fn_name,
                                        }).expect("Bad encoding");
                        let ty_id = abi::Call::type_id();
                        decoder.decode_type(ty_id, data);
                    }
```

- Which does bring up the fact that we're creating this `HashSet` on each `Receipt` match which is obviously not ideal (but outside of the scope of this PR). 
   - However I do think we should follow up on this to prevent us having to allocate this `HashSet` every indexer call #EveryComputationCounts 
   - Doesn't necessarily have to be you that does it

Reply:
Made a note of this. Will create an issue.